setup.py

```diff
@@ -35,21 +30,16 @@
     packages=["uharfbuzz"],
     zip_safe=False,
     setup_requires=["setuptools_scm"],
-    cmake_args=cmake_args,
     python_requires=">=3.5",
+    ext_modules = cythonize(Extension(
+        'uharfbuzz',
+        define_macros=[('HB_NO_MT', '1')],
```

Comment:
Guess HB_NO_MT is no issue here? Some more common compile flags would be nice, like HAVE_ROUNDF and things, guess you can use HB_TINY and things also.

Reply:
No `HB_TINY` please, we want as much of HarfBuzz API and functionality as possible, size does not matter here.

Reply:
Sure! Just wanted to indicate the place [your suggested configs](https://github.com/harfbuzz/uharfbuzz/issues/29#issuecomment-629475746) can be placed.

Reply:
What does HB_NO_MT do? I suppose MT stands for "multi-threaded"? Why disabling that for the python extension?

Reply:
Yes, it disables multi-threading support in HarfBuzz (i.e. makes it multi thread unsafe). Do Python extensions get called from different threads simultaneously, if not then HarfBuzz does not not need to be multi thread safe.

Reply:
> Why disabling that for the python extension?

This of course quickly is put together to show possibility of using amalgam for uharfbuzz, the reason that I've put it here is we should either declare we don't need MT support or we should detect which atomic/mutex things provider is available for the compile.

Using this simple to use python extension builder we can support #45 also just easily.

Reply:
I wonder whether HB_NO_MT was defined in the old cmake based setup as well? Or was cmake detecting that automatically? I'm ok to disable multi-threading support in uharfbuzz until we an explicit need for it. Cython allows to [release/reacquire the GIL](https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#releasing-the-gil) when running native code that doesn't interact with Python (so that other Python threads can run in parallel while native code is running). But you have to annotate that explicitly with `nogil` (we currently don't so we're holding the GIL and no MT is occurring).

Reply:
> I wonder whether HB_NO_MT was defined in the old cmake based setup as well?

CMake was able to detect that so most likely it wasn't the case there. But it is really isn't that uncommon thing, Firefox https://searchfox.org/mozilla-central/source/gfx/harfbuzz/src/moz.build#112 still uses harfbuzz with that config and Chrome was used to use HB_NO_MT.